### ISSUE TITLE
Why is the package creating a file in my homefolder?

### ISSUE THREAD

**Issue by davidanthoff** [JuliaGizmos/AssetRegistry.jl#13]
Why is this package putting a file `.jlassetregistry.json` into my home folder? That is really opaque, as a user it is entirely unclear where that comes from, what it is needed for, whether I can delete it etc. Really a package shouldn't just create files in the home folder.

**Comment by travigd** [comment]
Transferred to AssetRegistry.jl

There's no reason other than that's where it's gone historically. It could be changed but it needs to be in a known location since it is accessed by things outside of Julia (e.g. the WebIO Jupyter server extension).

I'm hesitant to change it, however, because it could lead to weird issues if using two versions of AssetRegistry.jl at the same time (e.g., the Jupyter server extension was configured with one location for the JSON file but the current Interact/WebIO/AssetRegistry versions loaded are using the old location).

**Comment by shashi** [comment]
Actually the reason was, our jupyter plugins need to be able to find this file (from Python). It would be awfully complicated to try to locate the right Julia package on disk without access to a running Julia process.

**Comment by travigd** [comment]
> Actually the reason was, our jupyter plugins need to be able to find this file (from Python). It would be awfully complicated to try to locate the right Julia package on disk without access to a running Julia process.

Yeah. There are probably _better_ places for it than in the root `$HOME` directory (e.g. `~/.config` or `~/.local` on Linux maybe?) but changing it is likely to produce weird errors and this package isn't super active to begin with.

**Comment by travigd** [comment]
Shouldn't this just be somewhere inside .julia? Other packages also put stuff there that they want to preserve across different package versions (e.g. Conda.jl).

I think that's a little bit different since Conda.jl has to store many things and this is a one-off file.

> I do think this should be updated. Software that just randomly puts files onto user systems really give a bad impression.

This is a subjective, mostly hypothetical concern with no easy migration path. Lots of software does this kind of thing, and while I agree that it's non-ideal, I think that the benefits are entirely aesthetic and it really does introduce the possibility for new bugs where there previously were none (and, again, this package really has no one to field those bugs if and when they arise).

**Comment by christopher-dG** [comment]
Here to add my support for keeping my home directory clean. This file should certainly be somewhere in a Julia depot! Otherwise, I will continue to delete it every time I see it. 😅

**Comment by travigd** [comment]
If you want to make a PR that addresses the concerns I enumerated above I’m more than happy to entertain it. 🤓

**Comment by christopher-dG** [comment]
I'll look into it 🙂

**Comment by travigd** [comment]
This will also need to be paired with a commit to WebIO to fix the jupyter extension I think.

**Comment by jwintersinger** [comment]
For what it's worth, `jlassetregistry.lock` is problematic in my environment. My university's compute cluster makes `$HOME` read-only on compute nodes, so anything that uses AssetRegistry.jl (namely PlotlyJS.jl) crashes. To accommodate Julia's precompile cache, I've set the `JULIA_DEPOT_PATH` environment variable to a writable location. Maybe AssetRegistry.jl could write to this location instead?
